test/e2e/framework/util.go

```diff
@@ -3825,19 +3825,38 @@ func RestartKubeProxy(host string) error {
 
 func RestartKubelet(host string) error {
 	// TODO: Make it work for all providers and distros.
-	if !ProviderIs("gce", "aws") {
+	if !ProviderIs("gce", "aws", "vsphere") {
 		return fmt.Errorf("unsupported provider: %s", TestContext.Provider)
```

Comment:
consider this slightly more informative error msg:
`fmt.Errorf("unsupported provider %q, supported providers are %s", TestContext.Provider, "gce, aws")`